eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/sink/handle/RetryHttpSinkHandler.java

```diff
@@ -0,0 +1,214 @@
+
+    private void initRetry() {
+        HttpRetryConfig httpRetryConfig = this.connectorConfig.getRetryConfig();
+
+        this.retryPolicyBuilder = RetryPolicy.<HttpResponse<Buffer>>builder()
+            .handleIf(e -> e instanceof ConnectException)
+            .handleResultIf(response -> httpRetryConfig.isRetryOnNonSuccess() && !HttpUtils.is2xxSuccessful(response.statusCode()))
+            .withMaxRetries(httpRetryConfig.getMaxRetries())
+            .withDelay(Duration.ofMillis(httpRetryConfig.getInterval()));
+    }
+
+
+    /**
+     * Initializes the WebClient for making HTTP requests based on the provided SinkConnectorConfig.
+     */
+    @Override
+    public void start() {
+        sinkHandler.start();
+    }
+
+
+    /**
+     * Processes a ConnectRecord by sending it over HTTP or HTTPS. This method should be called for each ConnectRecord that needs to be processed.
+     *
+     * @param record the ConnectRecord to process
+     */
+    @Override
+    public void handle(ConnectRecord record) {
+        for (URI url : this.urls) {
+            // convert ConnectRecord to HttpConnectRecord
+            String type = String.format("%s.%s.%s",
+                this.connectorConfig.getConnectorName(), url.getScheme(),
+                this.connectorConfig.getWebhookConfig().isActivate() ? "webhook" : "common");
+            HttpConnectRecord httpConnectRecord = HttpConnectRecord.convertConnectRecord(record, type);
+            // handle the HttpConnectRecord
+            deliver(url, httpConnectRecord);
+        }
+    }
+
+
+    /**
+     * Processes HttpConnectRecord on specified URL while returning its own processing logic This method provides the retry power to process the
+     * HttpConnectRecord
+     *
+     * @param url               URI to which the HttpConnectRecord should be sent
+     * @param httpConnectRecord HttpConnectRecord to process
+     * @return processing chain
+     */
+    @Override
+    public Future<HttpResponse<Buffer>> deliver(URI url, HttpConnectRecord httpConnectRecord) {
+        // Only webhook mode needs to use the firstTryId
+        String firstTryId = UUID.randomUUID().toString();
+
+        // Build the retry policy
+        RetryPolicy<HttpResponse<Buffer>> retryPolicy = retryPolicyBuilder
+            .onSuccess(e -> {
+                if (connectorConfig.getWebhookConfig().isActivate()) {
+                    // convert the result to an HttpExportRecord
+                    HttpExportRecord exportRecord = covertToExportRecord(e, e.getResult(), e.getException(), url, firstTryId);
+                    // add the data to the queue
+                    ((WebhookHttpSinkHandler) sinkHandler).addDataToQueue(exportRecord);
+                }
+            })
+            .onRetry(e -> {
+                if (log.isDebugEnabled()) {
+                    log.warn("Retrying the request to {} for the {} time. HttpConnectRecord= {}", url, e.getAttemptCount(), httpConnectRecord);
+                } else {
```

Comment:
`event -> {` instead of `e -> {` (exception) will be easier to be understood.

Reply:
OK. I will make the modification.

Reply:
Resolved at https://github.com/apache/eventmesh/pull/4837/commits/4af7455c9b09d4a1771fa7580b7406a61160ea74